include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -8945,6 +8945,16 @@ def err_bounds_type_annotation_lost_checking : Error<
 
   def err_checked_scope_no_variadic_func_for_expression : Error<
     "cannot use a variable arguments function in a checked scope or function">;
+
+  def err_bounds_cast_array_pointer_type_with_single_syntax
+      : Error<"invalid bounds cast : expected _Ptr or * type (%0)">;
+
+  def err_bounds_cast_integral_with_single_syntax
```

Comment:
Could you remove the space before the `:`?    You can also delete the `(%0)` because the error message will point directly at the problematic type.   The error messages for integral and array_pointer types are now identical.  I think you can just have one error message.